### PR TITLE
Fix behaviour around failed FF responses

### PR DESCRIPTION
**Pull request by rrpff**
The behaviour on a failed response is now to:
- try and extend the expiration of the last cached response if possible
- cache the failure if not
- but always refetch on load if the cached response is a failure

### CODE REVIEW

File: packages/tracking-sdk/src/flags-fetch.ts [hunk 1]
```diff
@@ -97,9 +97,17 @@ export async function fetchFlags(url: string, timeoutMs: number) {
...
       console.error("fetching flags: ", e);
     } finally {
-      // we cache even if the request failed to avoid having the UI
-      // wait and spam the API
-      cache.set(url, success, flags);
+      if (success) {
+        cache.set(url, success, flags);
+      } else if (cache.has(url)) {
+        // if there is a previous version, extend it's expireAt time
+        cache.extendExpiry(url);
```
Comment (roncohen) [inline packages/tracking-sdk/src/flags-fetch.ts:104]: drive-by comment, but could this just be a `get` followed by a `set`?

**Comment by roncohen** [comment]
could you rebase/merge main here?

**Review by matus-vacula — APPROVED** [review]
Can we move this forward? I still cannot see feature flags on prod

**Comment by makwarth** [comment]
Can we get this shipped? I can't see any FFs on prod

**Comment by makwarth** [comment]
Yay!
